curate/views_api.py

```diff
@@ -160,8 +160,23 @@ def list_articles(request):
     '''
     Return a list of all existing articles.
     '''
-    queryset=Article.objects.all().prefetch_related('commentaries')
-    serializer=ArticleListSerializer(instance=queryset, many=True)
+    PAGE_SIZE = 10
+    queryset = Article.objects.all().prefetch_related('commentaries')
+    serializer = ArticleListSerializer(instance=queryset, many=True)
+
+    paginator = PageNumberPagination()
+    paginator.page_size = PAGE_SIZE
+
+    result_page = paginator.paginate_queryset(queryset, request)
+
+    # Sort the results by field or property
+    ordering = request.query_params.get('ordering')
+    if ordering not in ['created', 'impact']:
+        ordering = 'created'
+    sorted_result_page = sorted(result_page, key=lambda article: getattr(article, ordering), reverse=True)
```

Comment:
Data needs to be sorted before pagination, not after. Use [order_by()](https://docs.djangoproject.com/en/2.2/ref/models/querysets/#django.db.models.query.QuerySet.order_by) to achieve this.

Reply:
Definitely needs to be sorted before pagination, d'oh!

I was sorting the results rather than using `order_by` because we need to sort by `impact` which is a model property rather than an actual database field. Looking at it again, it should be possible to do this with a [query expression](https://docs.djangoproject.com/en/2.2/ref/models/expressions/) which would be better than pulling in every article and sorting them. Thanks!